x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRerankingRankFeaturePhaseRankShardContext.java

```diff
@@ -69,8 +69,10 @@ public RankShardResult doBuildRankFeatureShardResult(SearchHits hits, int shardI
                         SemanticTextFieldMapper semanticTextFieldMapper = (SemanticTextFieldMapper) mapper;
                         SemanticTextFieldMapper.SemanticTextFieldType fieldType = semanticTextFieldMapper.fieldType();
 
-                        // We can't guarantee that all semantic_text embeddings will be compatible with indexed chunking settings,
-                        // so we take a hard line, scoring using BM25 if reranking on a semantic_text field with specified chunking_settings
+                        // We can't guarantee that the size of all semantic_text embeddings will match requested chunking settings,
```

Comment:
Super-nit:

```suggestion
                        // We can't guarantee that the size of all semantic_text chunks will match requested chunking settings,
```

We care about the size of the chunks, not the embeddings associated with each chunk. I know i'm being super nit-picky with this, but embeddings != chunks.